index.d.ts

```diff
@@ -0,0 +1,49 @@
+declare module '@articulate/authentic' {
+  interface VerifyOpts {
+    algorithms?: string[]
+    audience?: string | RegExp | Array<string | RegExp>
+    clockTimestamp?: number
+    clockTolerance?: number
+    issuer?: string | string[]
+    ignoreExpiration?: boolean
+    ignoreNotBefore?: boolean
+    jwtid?: string
+    subject?: string
+    maxAge?: string
+  }
+
+  interface JWKSOpts {
+    cache?: boolean
+    rateLimit?: boolean
+  }
+
+  interface AuthenticOpts {
+    issWhitelist: string[]
+    jwks?: JWKSOpts
+    verify?: VerifyOpts
+  }
+
+  interface Validator {
+    (token: string): Promise<Authentic.JWT>
+  }
+
+  function Authentic(opts: AuthenticOpts): Validator
+
+  namespace Authentic {
```

Comment:
Is there any reason for the namespace here?

Reply:
I had to use a namespace so we're able to use a namespace import

```js
import * as Authentic from "@articulate/authentic";
```

This allows users to still be able to do

```js
import * as Authentic from "@articulate/authentic"
Authentic() // function
Authentic.JWT // interface
```
```js
import authentic from "@articulate/authentic"
authentic() // function
```
```js
import { JWT } from "@articulate/authentic"
JWT // interface
```

Reply:
Without the namespace, you'll get an error in typscript: `Cannot invoke an expression whose type lacks a call signature.`